OpenTrons/opentrons-api
# feat(api): Do not enqueue json commands on protocol load

# Overview

closes https://opentrons.atlassian.net/browse/EXEC-352.
first step towards fixit commands. do not enqueue json protocol commands. 

# Test Plan

Tested with Json protocols and Postman:
- Make sure loading a protocol and executing it are happening within order.
- Make sure get run `/commands` returning the list properly with successful commands. 
- Make sure loading a failed protocol should fail the run and fail the command.
- Make sure get run `/commands` for failed runs the list of commands, last command being the failed command. 
- Fixed e2e test to comply with these changes


# Changelog

- Do no enqueue commands in PE for Json command upon load. 
- Execute commands one by one when run get started - same way we do for python protocols.


# Review requests

Changes make sense? 
GET run` /commands` will not return the full list of commands if the run did not start - its a change we are doing to make json protocols run like python protocols. are we ok with this? 

# Risk assessment

Medium. need to do smoke tests for Json protocols and make sure these changes do not affect anything.
